doc/sphinx/source/recipes/recipe_monitor.rst

```diff
@@ -0,0 +1,248 @@
+.. _recipe_monitor:
+
+Monitor
+#######
+
+Overview
+========
+
+Available recipes and diagnostics
+=================================
+
+Recipes are stored in `recipes/``
+
+  - recipe_monitor.yml
+
+Diagnostics are stored in `diag_scripts/monitor/`
+
+  - monitor.py:
+    Plots preprocessor output directly from the preprocessor.
+  - compute_eofs.py:
+    A sample on how to use the monitor structure to show other metrics.
+    Computes and plots the map of the first EOF and the associated PC timeseries.
+
+List of plot types available in monitor.py
+------------------------------------------
+
+- Climatology (plot_type `clim`): Plots climatology. Supported coordinates:
+  (`latitude`, `longitude`, `month_number`).
+
+- Seasonal climatologies (plot_type `seasonclim`): It produces a multi panel (2x2) plot
+  with the seasonal climatologies. Supported coordinates:
+  (`latitude`, `longitude`, `month_number`).
+
+- Monthly climatologies (plot_type `monclim`): It produces a multi panel (3x4) plot with
+  the monthly climatologies. Can be customized to show only certain months
+  and to rearrange the number of columns and rows. Supported coordinates:
+  (`latitude`, `longitude`, `month_number`).
+
+- Time series (plot_type `timeseries`): Generate time series plots. It will always
+  generate the full period time series, but if the period is longer than 75
+  years, it will also generate two extra time series for the first and last 50
+  years. It will produce multi panel plots for data with `shape_id` or `region`
+  coordinates of length > 1. Supported coordinates: `time`, `shape_id`
+  (optional) and `region` (optional).
+
+- Annual cycle (plot_type `annual_cycle`): Generate an annual cycle plot (timeseries
+  like climatological from January to December). It will produce multi panel
+  plots for data with `shape_id` or `region` coordinates of length > 1.
+  Supported coordinates: `time`, `shape_id` (optional) and `region` (optional).
+
+User settings
+=============
+
+User setting files are stored in recipes and in a dedicated yaml config file.
+
+In the variable definitions, users can set the attribute `plot_name` to fix
+the variable name that will be used for the plot's title. If it is not set,
+mapgenerator will try to choose a sensible one from the standard name
+attributes (`long_name`, `standard_name` and `var_name`)
+
+monitor.py
+----------
+
+  * plots:
+    a dictionary containing the plots to make, with its own options.
```

Comment:
>Really nice addition, thanks @sloosvel! I run the recipe successfully (also added other models from other projects). Some of the plots have overlapping text, but since this is highly customizable I don't think that's an issue now.

Javi's work mostly!

> I think it might be great to mention in the doc that this recipe can plot arbitrary model output. This is a feature that many users asked for. Maybe in the [FAQ]
(https://github.com/ESMValGroup/ESMValTool/blob/main/doc/sphinx/source/faq.rst)? Maybe something like "Can ESMValTool plot arbitrary model output?" and a link to this recipe?

Good idea, I will add that

>I noticed that some plots are in .svg format even though I use output-file-type=png. It looks like this is hardcoded in some parts of the diagnostics. What's the reason for that? I think we should stick to the format the user asks for.

Can't think of any reason, I guess is something pending to be cleaned. Will update.

Reply:
>> I noticed that some plots are in .svg format even though I use output-file-type=png. It looks like this is hardcoded in some parts of the diagnostics. What's the reason for that? I think we should stick to the format the user asks for.

>Can't think of any reason, I guess is something pending to be cleaned. Will update.

They are hardoced for size and looks reasons:

- `svg` can be zoomed in and out without losing quality, so it is prefered by default
- `svg` maps are enormous, so it had to be done in `png` format. Timeseries on the other hand are smaller in `svg` than `png`, so they were a better fit for the online app this was originally for.

Reply:
On the second point, probably [rasterization](https://matplotlib.org/devdocs/gallery/misc/rasterization_demo.html) would be a perfect fit.

Reply:
> On the second point, probably [rasterization](https://matplotlib.org/devdocs/gallery/misc/rasterization_demo.html) would be a perfect fit.

I think that is what is done when saving the files as `png`

Reply:
Well, when saving as png of course some form of rasterization takes place. But the point of the link and the associated feature in matplotlib is that you can selectively rasterize elements of a plot when storing in vector formats. This way, the raw map part is effectively embedded as a png or similar inside the svg or pdf, giving you small file sizes while retaining sharp labels and axis and whatnot at all zoom levels.

Reply:
Well in that case, for this PR maybe it's better to not modify the formats. This recipe and scripts are probably going to be expanded anyway, so details like that can also get addressed.

Reply:
svg literally means Scalable Vector Graphics :grin: Welcome back @jvegreg :beer:

Reply:
@valeriupredoi, yeah, I know all that. But what does it have to do with anything?

Reply:
> too little time

I do have very little time right now, sorry.

Reply:
@zklaus welcoming Javi, and offering support for your rasterization idea - but I reckon we should not dabble around this just here, I think this whole show will need quite a bit more work to be done in various follow-up PRs, so maybe we can discuss all this there - this is big and functional enough (alas, an excellent contribution!) to be merged now

Reply:
Understandable, @sloosvel. Could you open an issue about it?

Reply:
Done in #2561

Reply:
I took the liberty of assigning the 2.6.0 milestone :blush: